### ISSUE TITLE
[DATA-GRID]: sticky header conflicts with single-row selection

### ISSUE THREAD

**Issue by dvelluto** [Vonage/vivid-3#1412]
### Description

I've configured data-grid 
```
<style>
  html { 
    block-size: 300px; 
  }
</style>

<vwc-data-grid></vwc-data-grid>
<script>
	grid = document.querySelector('vwc-data-grid');
    grid.generateHeader = 'sticky'
    grid.selectionMode = 'single-row';
	grid.rowsData = [
			{data1: 'tabs', data2: 'will'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
	];
</script>
```
and:  
- sticky header stops to work if selectionMode = 'single-row'
- header is getting highlighted when mouse over it 

### Component(s) impacted

![image](https://github.com/Vonage/vivid-3/assets/57232860/c35330e4-1c1b-4022-a307-c79213ed3049)

Reproducible on https://vivid.deno.dev/components/data-grid 

### Browser

Microsoft Edge

### Vivid Web Components version

3.37.0

### Severity

Severity 2 = Aspects of design is broken, and impedes users in a significant way, but there is a way to complete their tasks. Affects major functionality, has a workaround.

### Application/website

VCC

### CodeSandbox example

https://vivid.deno.dev/

### Steps to reproduce the issue (if applicable)

go to https://vivid.deno.dev/ 
open data-grid section

paste in some code-boxes
```
<style>
  html { 
    block-size: 300px; 
  }
</style>

<vwc-data-grid></vwc-data-grid>
<script>
	grid = document.querySelector('vwc-data-grid');
    grid.generateHeader = 'sticky'
    grid.selectionMode = 'single-row';
	grid.rowsData = [
			{data1: 'tabs', data2: 'will'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
			{data1: 'not', data2: 'work'},
	];
</script>
```
observe that header is not sticky and select row is selecting header

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/vonage/vivid-3/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/vonage/vivid-3/issues) for duplicate issues

**Comment by dvelluto** [comment]
I just noticed that it is interfering also with the click / sort event if the header is also sortable.

**Comment by dvelluto** [comment]
ok, so I get that the onRowFocus event happening on the header row is intended, I can filter it in code checking `_rowType !== header`.

thanks for the ticket

**Comment by rachelbt** [comment]
> onRowFocus event happening on the header row is intended
I'm not sure - that it what is needed - this one need to be checked. But I'm happy that you have a fix for that anyway.

**Comment by rachelbt** [comment]
@dvelluto the sticky issue was fixed in vivid version 3.4.0.
I'm closing this issue.
Thank you for making Vivid better!
